recipes-core/systemd/systemd_%.bbappend

```diff
@@ -20,8 +22,41 @@ do_install_append() {
         install -m 1750 -d ${D}/mnt/data/journal
         chgrp systemd-journal ${D}/mnt/data/journal
         install -m 0644 ${WORKDIR}/var-log-journal.mount ${D}${systemd_system_unitdir}/
+
+        # (Re)create journal folder permissions in data partions, e.g. after a
+        # factory reset via tmpfiles.d.
+        # (https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html).
+        #
+        # We need that for scenarios where you update from an image without
+        # persistent journal to an image where persistent journal is enabled.
+        # Note: /mnt/data/journal is implicitly created by
+        #       'var-log-journal.mount'.
+        echo "z /mnt/data/journal 2750 root systemd-journal - -"                    >> ${D}${libdir}/tmpfiles.d/persistent_journal.conf
+        echo "z /mnt/data/journal/%m 2755 root systemd-journal - -"                 >> ${D}${libdir}/tmpfiles.d/persistent_journal.conf
+        echo "z /mnt/data/journal/%m/system.journal 0640 root systemd-journal - -"  >> ${D}${libdir}/tmpfiles.d/persistent_journal.conf
+
+        # Enable persistent machine-id creation only if journal is persistent
+        # Note: We can not simply create an etc-machine\x2did.mount unit.
+        #       Systemd creates a tmpfs mtab entry for /etc/machine-id which
+        #       prevents that we can mount /etc/machine-id with
+        #       "Where=" in etc-machine\x2did.mount.
+        #       Thus we do it as "ExecStartPre=" step in
+        #       systemd-machine-id-commit.service.
+        sed -i -E '/^ConditionPathIsMountPoint=(.*)/d' ${D}${systemd_system_unitdir}/systemd-machine-id-commit.service
+        sed -i -E 's?^ConditionPathIsReadWrite=(.*)?ConditionPathExists=!/mnt/etc/upper/machine-id?' ${D}${systemd_system_unitdir}/systemd-machine-id-commit.service
+        sed -i -E 's?^ExecStart=(.*)?ExecStartPre=touch /mnt/etc/upper/machine-id\nExecStartPre=mount -o bind /mnt/etc/upper/machine-id /etc/machine-id\nExecStart=/bin/systemd-machine-id-setup?' ${D}${systemd_system_unitdir}/systemd-machine-id-commit.service
     fi
+
+    # Start systemd-timesync after '/var/lib' is mounted rw.
+    # Even if it can not sync via ntp it will set the last saved time from
```

Comment:
Does this completely cover all edge cases we can have with devices without RTC?

Reply:
All I know of. We have a time as soon systemd-timesyncd starts. All logs until then have the timestamp of the kernel-build.